util/cron/test-gpu-rocm.bash

```diff
@@ -7,7 +7,6 @@ source $CWD/common-native-gpu.bash
 
 export CHPL_LLVM=bundled
 export CHPL_COMM=none
-export CHPL_GPU_CODEGEN=rocm
 export CHPL_LAUNCHER_PARTITION=amdMI60
 module load rocm
```

Comment:
Maybe I'm going to far but I almost want to have this script fail if:

`printchplenv --all --simple | grep -o "CHPL_GPU=[^ ]*" | grep -o "[^=]*$"` != "amd"

I won't push on it, but update the cuda script if you do decide to add this check here.

Reply:
Is your fear based on the possibility of this falling back to `cpu` or by mistake run on a system with an NVIDIA gpu? I can put the equivalent (`export CHPL_GPU=amd`) both here and in the CUDA script? It is probably a better practice to be explicit in these scripts anyway.

Reply:
Yeah my fear is we're relying on `CHPL_GPU` to be correctly inferred to `amd` and in theory (though maybe somewhat unlikely) this could regress without us noticing. My knee-jerk defensive programming instinct says "add an assertion".

You're right that adding `export CHPL_GPU=amd` would assuage that fear.

Though I suppose it's also nice to rely/demonstrate the implicit behavior of `chplenv` setting the sensible default and then adding in this kind of "assertion" to verify it. We don't have any other real test for this feature so it'd be kind of nice to have this as one (even if it's kind of a sledgehammery way of doing it).

Anyway, if you don't buy my reasoning I'd be fine with just explicitly setting `CHPL_GPU=amd` there and `CHPL_GPU=cuda` in the cuda script and moving on.

Reply:
I don't disagree. But I don't find that assertion necessary. I'll go with explicit `CHPL_GPU` setting.

We set `CHPL_LOCALE_MODEL=gpu` explicitly. After that, by default `CHPL_GPU` could go `nvidia`/`amd` or `none`. I am confident that we wouldn't choose the wrong vendor. If we did, explosion would be seen from space. We could fall back to `none` if we can't find nvcc or hipcc, in which case build will fail.

You have to set `CHPL_GPU` explicitly to `cpu` to enter the new mode, and I find it very unlikely to happen. It might be set in a script that you include, I guess. In that case, setting `CHPL_GPU` explicitly should be enough, I suppose.